ttyplot.c

```diff
@@ -289,9 +290,99 @@ void redraw_screen(const char * errstr) {
     refresh();
 }
 
+// Handle a single value from the input stream.
+// Return whether we got a full data record.
+int handle_value(double value)
+{
+    static double saved_value;
+    static int saved_value_valid = 0;
+
+    // First value of a 2-value record: save it for later.
+    if (two && !saved_value_valid) {
+        saved_value = value;
+        saved_value_valid = 1;
+        return 0;
+    }
+
+    // Otherwise we have a full record.
+    n = (n+1) % plotwidth;
+    if (two) {
+        values1[n] = saved_value;
+        values2[n] = value;
+        saved_value_valid = 0;
+    } else {
+        values1[n] = value;
+    }
+    if (rate)
+        td = derivative(&values1[n], two ? &values2[n] : NULL, &now);
+    return 1;
+}
+
+// Parse the input data. Return the number of bytes consumed.
+size_t parse_input(char *buffer, size_t length)
+{
+    static const char delimiters[] = " \t\r\n";  // white space
+
+    // Find the last delimiter.
+    int end;
+    for (end = length - 1; end >= 0; end--) {
+        if (strchr(delimiters, buffer[end]))  // found
+            break;
+    }
```

Comment:
This seems to find the rightmost delimiter. Can we leverage `strcspn` here to simplify? if not, this could be extracted as a function with a good name.

Reply:
`strcspn` wouldn't work: it scans from the beginning of the string, whereas we want to find the _last_ delimiter. What we need is a version of `strpbrk` that works in reverse, scanning from the end of the string.

I hesitated to pull this out into its own function. But it does make sense to do so: that could have been a function of `string.h`, maybe named `strnrpbrk` according to the current naming convention. I made the function, but went for a different name.

Reply:
@edgar-bonet extracting function `find_last` was a good move, very nice, thanks!

Reply:
PS: I was too quick with `strcspn`, should have had a closer look. Maybe it was wishful thinking.